presentation/backend/src/presentation-backend/RulesetEmbedder.ts

```diff
@@ -17,16 +18,48 @@ import {
 import { Ruleset } from "@bentley/presentation-common";
 import { PresentationRules } from "./domain/PresentationRulesDomain";
 import * as RulesetElements from "./domain/RulesetElements";
+import { isEnum, normalizeVersion } from "./Utils";
 
 /**
- * Available strategies for handling duplicate rulesets.
+ * Strategies for handling duplicate rulesets.
  * @beta
+ * @deprecated Use [[RulesetInsertOptions]]
  */
 export enum DuplicateRulesetHandlingStrategy {
+  /** Do not insert the ruleset if another ruleset with the same ID already exists. */
   Skip,
+
+  /** Replace already existing ruleset if one exists with the same ID and version. */
   Replace,
 }
 
+/**
+ * Options for [[RulesetEmbedder.insertRuleset]] operation.
+ * @beta
+ */
+export interface RulesetInsertOptions {
+  /**
+   * When should insertion be skipped:
+   * - `same-id` - if iModel already contains a ruleset with the same id and **any** version
+   * - `same-id-and-version-eq` - if iModel already contains a ruleset with same id and version
+   * - `same-id-and-version-gte` - if iModel already contains a ruleset with same id and
+   * version equal to greater than of the inserted ruleset.
+   *
+   * Defaults to `same-id-and-version`.
+   */
+  skip?: "never" | "same-id" | "same-id-and-version-eq" | "same-id-and-version-gte";
+
+  /**
+   * Which existing versions of rulesets with same id should be replaced when we insert a new one:
+   * - `all` - replace all rulesets with same id.
+   * - `all-lower` - replace rulesets with same id an version lower than the version of inserted ruleset.

```

Comment:
Probably a typo in line 55, `an` -> `and`.

Reply:
fixed